src/NuGetForUnity/Editor/NugetWindow.cs

```diff
@@ -724,23 +723,27 @@ private void DrawUpdatesHeader()
                         UpdateUpdatePackages();
                     }
 
-                    if (GUILayout.Button("Install All Updates", GUILayout.Width(150)))
+                    if (updatePackages.Count > 0)
                     {
-                        NugetHelper.UpdateAll(updatePackages, NugetHelper.InstalledPackages);
-                        NugetHelper.UpdateInstalledPackages();
-                        UpdateUpdatePackages();
-                    }
+                        if (GUILayout.Button("Update All", GUILayout.Width(150)))
+                        {
+                            NugetHelper.UpdateAll(updatePackages, NugetHelper.InstalledPackages);
+                            NugetHelper.UpdateInstalledPackages();
+                            UpdateUpdatePackages();
+                        }
 
-                    if (GUILayout.Button("Refresh", GUILayout.Width(60)))
-                    {
-                        Refresh(true);
+                        if (updatePackages.Any(selectedPackages.Contains))
+                        {
+                            if (GUILayout.Button("Update Selected", GUILayout.Width(200)))
+                            {
+                                NugetHelper.UpdateAll(updatePackages.Where(p => selectedPackages.Contains(p)), NugetHelper.InstalledPackages);
```

Comment:
```suggestion
                                NugetHelper.UpdateAll(updatePackages.Where(selectedPackages.Contains), NugetHelper.InstalledPackages);
```
It's the same but just for "style" 😄 .